sympy/parsing/latex/lark/transformer.py

```diff
@@ -0,0 +1,416 @@
+import re
+
+import sympy
+from sympy.parsing.latex.errors import LaTeXParsingError
+
+import lark
+
+# noinspection PyPep8Naming,PyMethodMayBeStatic
+class TransformToSymPyExpr(lark.Transformer):
+    SYMBOL = sympy.Symbol
+    DIGIT = sympy.core.numbers.Integer
+
+    def GREEK_SYMBOL(self, tokens):
+        # we omit the first character because it is a backslash. Also, if the variable name has "var" in it,
+        # like "varphi" or "varepsilon", we remove that
+        variable_name = re.sub("var", "", tokens[1:])
+        if variable_name == "lambda":
+            # we do the same name change as sympy.abc because lambda is a Python keyword
+            return sympy.Symbol("lamda")
+        else:
+            return sympy.Symbol(variable_name)
+
+    def BASIC_SUBSCRIPTED_SYMBOL(self, tokens):
+        symbol, sub = tokens.value.split("_")
+        if sub.startswith("{"):
+            return sympy.Symbol("%s_{%s}" % (symbol, sub[1:-1]))
+        else:
+            return sympy.Symbol("%s_{%s}" % (symbol, sub))
+
+    def GREEK_SUBSCRIPTED_SYMBOL(self, tokens):
+        greek_letter, sub = tokens.value.split("_")
+
+        greek_letter = re.sub("var", "", greek_letter[1:])
+        if greek_letter == "lambda":
+            # we do the same name change as sympy.abc because lambda is a Python keyword
+            greek_letter = "lamda"
+
+        if sub.startswith("{"):
+            return sympy.Symbol("%s_{%s}" % (greek_letter, sub[1:-1]))
+        else:
+            return sympy.Symbol("%s_{%s}" % (greek_letter, sub))
+
+    def SYMBOL_WITH_GREEK_SUBSCRIPT(self, tokens):
+        symbol, sub = tokens.value.split("_")
+        if sub.startswith("{"):
+            greek_letter = sub[2:-1]
+            greek_letter = re.sub("var", "", greek_letter)
+            if greek_letter == "lambda":
+                # we do the same name change as sympy.abc because lambda is a Python keyword
+                greek_letter = "lamda"
```

Comment:
I'm not sure that `lambda` is a invalid Symbol name
`Symbol('lambda')` works.

Reply:
Like the comment says, I simply mimicked `sympy.abc`'s approach here. If you think that using `lambda` will not cause issues, then I am fine with making the change.

Reply:
I think we should keep it that way because if the user tries to use `subs` with an expression that has a lambda in it, the code will not work as intended.

Reply:
I believe the symbol name should be lambda, not lamda.

Reply:
I've made the change.